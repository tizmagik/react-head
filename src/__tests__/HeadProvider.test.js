import * as React from 'react';
import TestRenderer from 'react-test-renderer';

jest.setMock('react-dom', {
  createPortal: children => <>{children}</>,
});

describe('HeadProvider', () => {
  const { HeadProvider, Style } = require('../');

  it('adds HeadTags to given array from component tree', () => {
    const arr = [];
    TestRenderer.create(
      <HeadProvider headTags={arr}>
        <div>
          <Style name="name1" another="value1" />
          <Style test="test2" third="value2" />
        </div>
      </HeadProvider>
    );

    expect(arr.length).toBe(2);
    expect(arr).toMatchSnapshot();
  });
});
