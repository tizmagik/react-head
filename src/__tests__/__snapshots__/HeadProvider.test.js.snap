// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeadProvider adds HeadTags to given array from component tree 1`] = `
Array [
  <style
    another="value1"
    data-rh=""
    name="name1"
  />,
  <style
    data-rh=""
    test="test2"
    third="value2"
  />,
]
`;
